### PR TITLE
Added TextBox control and sample

### PR DESCRIPTION
**Pull request by ConradRoesch**

### CODE REVIEW

File: src/Fabulous.Avalonia/Widgets/Controls/TextBox.fs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    let VerticalContentAlignment = Attributes.defineAvaloniaPropertyWithEquality TextBox.VerticalContentAlignmentProperty
+
+    let AcceptsReturn = Attributes.defineAvaloniaPropertyWithEquality TextBox.AcceptsReturnProperty
+    
+    let AcceptsTab = Attributes.defineAvaloniaPropertyWithEquality TextBox.AcceptsTabProperty
+    
+    let IsReadOnly = Attributes.defineAvaloniaPropertyWithEquality TextBox.IsReadOnlyProperty
+
+    let IsUndoEnabled = Attributes.defineAvaloniaPropertyWithEquality TextBox.IsUndoEnabledProperty
+
```
Comment (edgarfgp) [inline src/Fabulous.Avalonia/Widgets/Controls/TextBox.fs:32]: Can we also add a wrapper for: 
- UndoLimit
- SelectionStart
- SelectionEnd

File: src/Fabulous.Avalonia/Widgets/Controls/TextBox.fs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    // Property Widgets
+    let CaretBrush = Attributes.defineAvaloniaPropertyWidget TextBox.CaretBrushProperty
+
+    let SelectionBrush = Attributes.defineAvaloniaPropertyWidget TextBox.SelectionBrushProperty
+   
+    let SelectionForegroundBrush = Attributes.defineAvaloniaPropertyWidget TextBox.SelectionForegroundBrushProperty
+
+    // Events
+    let ValueChanged = Attributes.defineAvaloniaPropertyWithChangedEvent' "TextBox_ValueChanged" TextBox.TextProperty
+
```
Comment (edgarfgp) [inline src/Fabulous.Avalonia/Widgets/Controls/TextBox.fs:69]: Can we also add event wrappers for :
- CopyingToClipboardEvent
- PastingFromClipboardEvent

You can use **Attributes.defineEvent**
ie:
```fsharp
let CopyingToClipboardEvent =
        Attributes.defineEvent<RoutedEventArgs> "TextBox_CopyingToClipboardEvent" (fun target ->
            (target :?> TextBox).CopyingToClipboard)
```

**Review by edgarfgp — CHANGES_REQUESTED** [review]
Awesome. Thanks for this :). I just added couple of comments

**Comment by ConradRoesch** [comment]
Thank you for the feedback and code formatting tips. I should be be able to get to those missing wrappers shortly. Are there any particular controls or other developments that you could use some help with in the near term?

**Comment by edgarfgp** [comment]
> Thank you for the feedback and code formatting tips. I should be able to get to those missing wrappers shortly. Are there any particular controls or other developments that you could use some help with in the near term?

I'm creating a list of controls to help us with the missing ones. I will create an issue to track this later today :)

**Comment by edgarfgp** [comment]
I’m merging this as it is and work on the playground app to help us with the testing . For now you can testing it using the counter app but not commit the changes 😀

**Review by edgarfgp — APPROVED** [review]

**Pull request merged**